pkg/pillar/cmd/zedmanager/zedmanager.go

```diff
@@ -412,6 +445,54 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject, ar
 	}
 }
 
+func handleLocalAppInstanceConfigCreate(ctx interface{}, key string, config interface{}) {
+	log.Noticef("handleLocalAppInstanceConfigCreate(%s)", key)
+	zedmanagerCtx := ctx.(*zedmanagerContext)
+	localConfig := config.(types.AppInstanceConfig)
+	oldConfig := lookupAppInstanceConfig(zedmanagerCtx, localConfig.Key(), false)
+	if oldConfig == nil {
+		log.Fatalf("handleLocalAppInstanceConfigCreate: no regular AppInstanceConfig for %s", key)
+	}
+	handleModify(ctx, key, localConfig, *oldConfig)
+}
+
+func restoreAvailableSnapshots(aiStatus *types.AppInstanceStatus) {
+	// List all the directories that are present in snapshots directory
+	// and restore the snapshot status for each of them
+	snapDir := types.SnapshotsDirname
+	dirEntries, err := os.ReadDir(snapDir)
+	if err != nil {
+		log.Warnf("No %s directory, nothing to restore", snapDir)
+		return
+	}
+	for _, dirEntry := range dirEntries {
+		if !dirEntry.IsDir() {
+			continue
+		}
+		snapshotID := dirEntry.Name()
+		// Figure out the ID of the app that this snapshot belongs to
+		aiConfig := deserializeAppInstanceConfigFromSnapshot(snapshotID)
+		if aiConfig == nil {
+			log.Warnf("cannot deserialize config for snapshot %s", snapshotID)
+			continue
+		}
+		if aiConfig.UUIDandVersion.UUID != aiStatus.UUIDandVersion.UUID {
+			// This snapshot is not for this app
```

Comment:
Shouldn't there be a log.Warning here? This should never happen AFAICT.

Reply:
No, it's expected behavior. On the system boot, after restart EVE goes through the list of apps, and for each of them, it looks for available snapshots serialized. If the snapshot parsed is not for this particular app - EVE skips it and goes further.

Reply:
OK